.meta/dietpi-imager

```diff
@@ -87,7 +87,14 @@
 			fi
 
 			# Visually separate dev name and size and add model and serial
-			for ((i=1;i<${#G_WHIP_MENU_ARRAY[@]};i+=2)); do G_WHIP_MENU_ARRAY[$i]=": $(lsblk -drno SIZE,MODEL,SERIAL "${G_WHIP_MENU_ARRAY[$i-1]}")"; done
+			for ((i=1;i<${#G_WHIP_MENU_ARRAY[@]};i+=2))
+			do
+				G_WHIP_MENU_ARRAY[$i]="$(lsblk -drno SIZE,MODEL,SERIAL "${G_WHIP_MENU_ARRAY[$i-1]}")"
+				G_WHIP_MENU_ARRAY[$i-1]+=':'
+			done
+
+			# Sort array
+			G_WHIP_MENU_ARRAY=($(tr ' ' '\n' <<< "${G_WHIP_MENU_ARRAY[@]}" | mawk '!seen[$0]++' | tr '\n' ' '))
```

Comment:
Indeed, the sort is misplaced there, removing all but one "ext4" line and breaking the intended array composition. May I suggest a simpler solution:
```
mapfile -t G_WHIP_MENU_ARRAY < <(mawk -v root="$(lsblk -npo PKNAME "$G_ROOTFS_DEV")" '{if ( $1!=root && $2=="ext4" ) print $1"\n"$1"_details"}' < <(lsblk -rnpo PKNAME,FSTYPE) | sort -u)
```
When PKNAME = $1 is a duplicate, which is what we want to sort out, then <PKNAME>_details is a duplicate as well, so both are sorted out together. The second field/line is a only a placeholder to replace with more helpful information, as in bash one cannot insert array entries at a specific index and having all later entries moved one index upwards.

Probably a separate temporary array would be cleaner that contains only PKNAME entries, removes duplicates, and the second loop then creates G_WHIP_MENU_ARRAY from scratch from temporary array + details. This placeholder method looks a bit hacky 😄.

Reply:
That's much better, I was trying to sort the array after adding the extra details.